### PR TITLE
MAHOUT-1838: Simple 2d plotting of a sampled DRM

### PR DESCRIPTION
**Pull request by andrewpalumbo**
This is a bare-bones beginning to some mahout plots.  Takes a `DRM` and creates a scatter plot of a sample of the first two columns.

### CODE REVIEW

File: math-scala/pom.xml [hunk 1]
```diff
@@ -140,6 +140,13 @@
       <version>3.3.0</version>
     </dependency>
 
+
+    <dependency>
+      <groupId>com.github.haifengl</groupId>
+      <artifactId>smile-plot</artifactId>
+      <version>1.1.0</version>
+    </dependency>
+
```
Comment (andrewpalumbo) [inline math-scala/pom.xml]: @pferrel this simple integration only requires the single smile-plot artifact.

File: examples/bin/spark-shell-plot.mscala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+val drmSin = drmRand.mapBlock() {case (keys, block) =>  
+  val blockB = block.like()
+  for (i <- 0 until block.nrow) {
+    blockB(i, 0) = block(i, 0) 
+    blockB(i, 1) = Math.sin((block(i, 0) * 8))
+  }
+  keys -> blockB
+}
+
+new mplot2d(drmRand, samplePercent = 10)
```
Comment (smarthi) [inline examples/bin/spark-shell-plot.mscala]: make the samplePercent user defined ??

File: math-scala/src/main/scala/org/apache/mahout/visualization/mplot2d.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+/**
+  * Create a s scatter plot of a DRM by sampling a given percentage
+  * and plotting corresponding points of (drmXY(::,0),drmXY(::,2))
+  *
+  * @param drmXY a m x 2 Drm drm to plot
+  * @param samplePercent the percentage the drm to sample
+  * @tparam K
+  */
+class mplot2d[K](drmXY: DrmLike[K], samplePercent: Int = 10, setVisible: Boolean = true)  {
+     val drmSize = drmXY.checkpoint().numRows()
```
Comment (andrewpalumbo) [inline math-scala/src/main/scala/org/apache/mahout/visualization/mplot2d.scala]: @smarthi sample percent is user defined here.

File: examples/bin/spark-shell-plot.mscala (hunk 2 above)
Comment (andrewpalumbo) [inline examples/bin/spark-shell-plot.mscala]: that is just for illistrituve purposes,  the default is also 10% but the user can override it to be 1 to 100 %

Comment (andrewpalumbo) [inline examples/bin/spark-shell-plot.mscala]: changed the default sample size to 1%

**Comment by andrewpalumbo** [comment]
This should be pretty much good to go as a beginning as is.  As I said it is very bare-bones,  No titles, single plots, etc, but it is a starting point.

**Comment by smarthi** [comment]
+1 to merge

**Pull request closed without merging**